### ISSUE TITLE
sorting works in a weird way

### ISSUE THREAD

**Issue by wxcafe** [iocage/iocage#482]
```
15:56:03 < yoshi ansible > sudo iocage list -s ip4
+-----+----------+-------+--------------+--------------+
| JID |   NAME   | STATE |   RELEASE    |     IP4      |
+=====+==========+=======+==============+==============+
| 1   | smtpd    | up    | 11.1-RELEASE | 192.168.1.10 |
+-----+----------+-------+--------------+--------------+
| 10  | dl       | up    | 11.1-RELEASE | 192.168.1.90 |
+-----+----------+-------+--------------+--------------+
| 2   | www      | up    | 11.1-RELEASE | 192.168.1.20 |
+-----+----------+-------+--------------+--------------+
| 3   | xmpp     | up    | 11.1-RELEASE | 192.168.1.30 |
+-----+----------+-------+--------------+--------------+
| 4   | dns      | up    | 11.1-RELEASE | 192.168.1.40 |
+-----+----------+-------+--------------+--------------+
| 5   | imap     | up    | 11.1-RELEASE | 192.168.1.50 |
+-----+----------+-------+--------------+--------------+
| 6   | resolver | up    | 11.1-RELEASE | 192.168.1.53 |
+-----+----------+-------+--------------+--------------+
| 7   | cal      | up    | 11.1-RELEASE | 192.168.1.60 |
+-----+----------+-------+--------------+--------------+
| 8   | vpn      | up    | 11.1-RELEASE | 192.168.1.70 |
+-----+----------+-------+--------------+--------------+
| 9   | www-php  | up    | 11.1-RELEASE | 192.168.1.80 |
+-----+----------+-------+--------------+--------------+
15:56:07 < yoshi ansible > iocage --version
Version	0.9.10 2017/12/22
15:56:56 < yoshi ansible > sudo iocage list -s release
+-----+----------+-------+--------------+--------------+
| JID |   NAME   | STATE |   RELEASE    |     IP4      |
+=====+==========+=======+==============+==============+
| 4   | dns      | up    | 11.1-RELEASE | 192.168.1.40 |
+-----+----------+-------+--------------+--------------+
| 7   | cal      | up    | 11.1-RELEASE | 192.168.1.60 |
+-----+----------+-------+--------------+--------------+
| 9   | www-php  | up    | 11.1-RELEASE | 192.168.1.80 |
+-----+----------+-------+--------------+--------------+
| 10  | dl       | up    | 11.1-RELEASE | 192.168.1.90 |
+-----+----------+-------+--------------+--------------+
| 1   | smtpd    | up    | 11.1-RELEASE | 192.168.1.10 |
+-----+----------+-------+--------------+--------------+
| 5   | imap     | up    | 11.1-RELEASE | 192.168.1.50 |
+-----+----------+-------+--------------+--------------+
| 2   | www      | up    | 11.1-RELEASE | 192.168.1.20 |
+-----+----------+-------+--------------+--------------+
| 3   | xmpp     | up    | 11.1-RELEASE | 192.168.1.30 |
+-----+----------+-------+--------------+--------------+
| 8   | vpn      | up    | 11.1-RELEASE | 192.168.1.70 |
+-----+----------+-------+--------------+--------------+
| 6   | resolver | up    | 11.1-RELEASE | 192.168.1.53 |
+-----+----------+-------+--------------+--------------+
```

the ip4 sorting sorts by JID (as far as I can tell)
the release sorting is totally random, I expected it to sort by alphabetical in a given release

- [x] Supply `iocage --version`
- [x] Supply the commands used, along with any steps to recreate it.
- [x] Provide the output from the command you issued.
- [x] Supply what you expected the result or output to be
- [x] Checked that the problem has not already been fixed on `master` if using
 a stable release.

**Comment by wxcafe** [comment]
(I'm also of the opinion that sorting by JID should sort 10 after 9, but that's another problem I guess)

**Comment by jsegaert** [comment]
Both `iocage list -s release` and `iocage list -l -s release` result in a traceback in master: 

```
/home/jurgen jurgen@trinity%  iocage list -s release
Traceback (most recent call last):
  File "/usr/local/bin/iocage", line 10, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/iocage/cli/list.py", line 81, in cli
    dataset_type, header, _long, _sort, plugin=plugins, quick=quick)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/iocage.py", line 1144, in list
    exit_on_error=self.exit_on_error).list_datasets()
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_list.py", line 76, in list_datasets
    _all = self.list_all(ds)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_list.py", line 271, in list_all
    jail_list.sort(key=sort)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_common.py", line 320, in sort_release
    ordered_r_dict = collections.OrderedDict(sorted(r_dict.items()))
TypeError: '<' not supported between instances of 'float' and 'str'

/home/jurgen jurgen@trinity%  iocage list -l -s release
Traceback (most recent call last):
  File "/usr/local/bin/iocage", line 10, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/iocage/cli/list.py", line 81, in cli
    dataset_type, header, _long, _sort, plugin=plugins, quick=quick)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/iocage.py", line 1144, in list
    exit_on_error=self.exit_on_error).list_datasets()
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_list.py", line 76, in list_datasets
    _all = self.list_all(ds)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_list.py", line 271, in list_all
    jail_list.sort(key=sort)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_common.py", line 320, in sort_release
    ordered_r_dict = collections.OrderedDict(sorted(r_dict.items()))
TypeError: '<' not supported between instances of 'str' and 'float'
```

**Comment by jsegaert** [comment]
@skarekrow  These issues were addressed with PR #582 so this issue can be closed.
